test/localstack_test.py

```diff
@@ -14,16 +14,16 @@ def test_ec2_lifecycle_with_local_stack(local_stack, default_asset_id):
     """
     print("run ec2_setup!")
     execution_generator = run_lifecycle_for_ec2(AwsLocalStackAccess(None), None, None, None, default_asset_id.tag_value)
-    res = next(execution_generator)
-    while res[0] == "pending":
-        res = next(execution_generator)
+    ec2_data = next(execution_generator)
+    ec2_instance_description, key_file_location = ec2_data
+    while ec2_instance_description.is_pending:
+        ec2_data = next(execution_generator)
+        ec2_instance_description, key_file_location = ec2_data
```

Comment:
extract to function

Reply:
As discussed offline, the test is not so big,  I believe extracting the polling to a function would make the test harder to read.